src/templates/base/2019/base.html

```diff
@@ -99,11 +99,134 @@
+  </label>
+  <select id="language-switcher-{{switcher_name}}">
+    <option disabled selected value> {{ self.table_of_contents_title() }} </option>
+    <option value="{{ url_for('table_of_contents', year=year, lang=lang) }}">{{ self.foreword_title() }}</option>
+    {% for part_config in config.outline %}
+    {% for chapter_config in part_config.chapters %}
+      {% set title = localizedChapterTitles[chapter_config.slug] if localizedChapterTitles[chapter_config.slug]|length else chapter_config.title %}
+      {% if chapter_config.todo %}
+        <option disabled>
+          {{ self.chapter() }} {{ chapter_config.chapter }}: {{title}}
+        </option>
+      {% else %}
+        {% if chapter_lang_exists(lang, year, chapter_config.slug) %}
+          {% if current_title == title %}
+            <option selected value="{{ url_for('chapter', year=year, lang=lang, chapter=chapter_config.slug) }}">
+              {{ self.chapter() }} {{ chapter_config.chapter }}: {{title}}
+            </option>
+          {% else %}
+            <option value="{{ url_for('chapter', year=year, lang=lang, chapter=chapter_config.slug) }}">
+              {{ self.chapter() }} {{ chapter_config.chapter }}: {{title}}
+            </option>
+          {% endif %}
+        {% else %}
+        {% if current_title == title %}
+          <option selected value="{{ url_for('chapter', year=year, lang='en', chapter=chapter_config.slug) }}">
+            {{ self.chapter() }} {{ chapter_config.chapter }}: {{title}}
+          </option>
+        {% else %}
+          <option value="{{ url_for('chapter', year=year, lang='en', chapter=chapter_config.slug) }}">
+            {{ self.chapter() }} {{ chapter_config.chapter }}: {{title}}
+          </option>
+        {% endif %}
+        {% endif %}
+      {% endif %}
+    {% endfor %}
+    {% endfor %}
+
+    <option value="{{ url_for('methodology', year=year, lang=lang) }}">
+      {{ self.methodology_title() }}
+    </option>
+    <option value="{{ url_for('contributors', year=year, lang=lang) }}">
+      {{ self.contributors_title() }}
+    </option>
+    <option value="/static/pdfs/web_almanac_{{ year }}_{{ lang }}.pdf">
+      {{ self.ebook_title() }}
+    </option>
+  </select>
+</div>
+{% endmacro %}
+
+{% macro styled_table_of_contents_dropdown(switcher_name) %}
+{% set current_title = metadata.get('title') if metadata else None %}
+<div class="nav-dropdown {{switcher_name}} table-of-contents">
+  <button class="nav-dropdown-btn enable-script" aria-expanded="false" aria-label="{{ self.table_of_contents_title() }}" >
+    <a class="js-enable-script-remove-tag" href="{{ url_for('table_of_contents', year=year, lang=lang) }}">{{ self.table_of_contents_title() }}</a>
+  </button>
+  <ul class="nav-dropdown-list hidden {{switcher_name}}-list">
+    
+    <li class="nav-dropdown-list__chapter foreword">
+      <a href="{{ url_for('table_of_contents', year=year, lang=lang) }}">{{ self.foreword_title() }}</a>
+    </li>
```

Comment:
The foreword looks a little lonely without a heading. What do you think about this:

```suggestion
    <li class="nav-dropdown-list__part">
      {{ self.introduction() }}
    </li>
    <li class="nav-dropdown-list__chapter">
      {% if request.path.endswith("table-of-contents") %}
      <span class="nav-dropdown-list__current">{{ self.foreword_title() }}</span>
      {% else %}
      <a href="{{ url_for('table_of_contents', year=year, lang=lang) }}">{{ self.foreword_title() }}</a>
      {% endif %}
    </li>
```

![Addition of Introduction header in menu](https://user-images.githubusercontent.com/10931297/97902850-12748e00-1d36-11eb-9976-89f8c92b56ca.png)

Reply:
How about "Foreword" itself being the header?

Reply:
And what as the item? "Table of Contents"? Don't like the idead of making the header clickable as others aren't so need an item IMHO.

Reply:
No item IMO. Just as it's ok to have an H2 with no H3 in the outline sidebar, this component could have headings with no items.

Reply:
OK but then the heading will be a link. But "Part 1. Page content" will not be a link - and neither will any of the other headers. Isn't that inconsistent and confusing?

Reply:
We could make Part X headings link to the first chapter in their section too for consistency. Otherwise I think its ok.

Reply:
That's not a bad idea. Wasn't that happy with the headings not being links as kind of similar in style to the items.

Could also link to here: https://almanac.httparchive.org/en/2019/table-of-contents#part-1

Will have a think. Let's come back to this one and address the other issues first.

Reply:
Yeah deep linking to the part in the TOC sounds like a good idea to me. 👍

Reply:
ok i fixed.